src/librustc_mir/build/expr/category.rs

```diff
@@ -50,14 +50,12 @@ impl Category {
             ExprKind::Index { .. } |
             ExprKind::SelfRef |
             ExprKind::VarRef { .. } |
-            ExprKind::StaticRef { .. } =>
-                Some(Category::Lvalue),
+            ExprKind::StaticRef { .. } => Some(Category::Lvalue),
```

Comment:
I guess it's ok because there's a blank line, but in these cases, I follow a pretty strict rule such that I can easily identify groups of `|` patterns -- basically there needs to be at least a blank line or some indentation separating the groups.

In other words, I want to avoid this:

```rust
match foo {
   Bar |
   Baz => expr,
   Something |
   SomethingElse => expr2,
}
```

I find it hard to distinguish the various cases there, it's easy not to notice the `=>`.